coresdk/src/backend/web_server_driver.cpp

```diff
@@ -0,0 +1,122 @@
+//
+//  web_server.cpp
+//  splashkit
+//
+//  Created by James Armstrong on 21/08/2016.
+//  Copyright © 2016 Andrew Cain. All rights reserved.
+//
+
+#include "web_server_driver.h"
+
+#include "blockingconcurrentqueue.h"
+
+#include <iostream>
+#include <cstring>
+
+static sk_server_request* last_request = nullptr;
+static moodycamel::ConcurrentQueue<sk_server_request*> request_queue;
+
+static int begin_request_handler(struct mg_connection *conn)
+{
+    const struct mg_request_info *request_info = mg_get_request_info(conn);
+
+    sk_server_request *r = new sk_server_request;
+    r->uri = request_info->request_uri;
+
+    request_queue.enqueue(r);
+    r->control.acquire(); // Waits until user returns response.
+
+    char content[100];
+
+    int content_length = snprintf(content, sizeof(content),
+                                  r->response->message.c_str());
+
+    // Send HTTP reply to the client
+    mg_printf(conn,
+              "HTTP/1.1 200 OK\r\n"
+              "Content-Type: text/plain\r\n"
+              "Content-Length: %d\r\n" // Always set Content-Length
+              "\r\n"
+              "%s",
+              content_length, content);
```

Comment:
Could you just use `r->response->length()` and `r->response->c_str()` here and remove content and content_length?